src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -718,7 +718,7 @@ void NotationViewInputController::mouseMoveEvent(QMouseEvent* event)
     }
 
     // move canvas
-    if (!isNoteEnterMode) {
+    if ((!isNoteEnterMode) || (event->buttons() & Qt::MiddleButton)){
```

Comment:
Please create a new variable isMiddleButton and reuse it here and in 683 line

Reply:
also, fix code style

```
-    if ((!isNoteEnterMode) || (event->buttons() & Qt::MiddleButton)){
+    if (!isNoteEnterMode || isMiddleButton) {
```

Reply:
```bool isMiddleButton  = (event->buttons() & Qt::MiddleButton);```
```if  (!isNoteEnterMode || isMiddleButton) {
        m_view->moveCanvas(dragDelta.x(), dragDelta.y());
        m_isCanvasDragged = true;
    }```
can you please check this.

Reply:
bool isMiddleButton  = (event->buttons() & Qt::MiddleButton);

Reply:
```
bool isMiddleButton  = (event->buttons() & Qt::MiddleButton);
```
```
 // move canvas
    if (!isNoteEnterMode || isMiddleButton) {
        m_view->moveCanvas(dragDelta.x(), dragDelta.y());
        m_isCanvasDragged = true;
    }
```
can you please check this.

Reply:
```
bool isMiddleButton  = (event->buttons() & Qt::MiddleButton);
bool isDragObjectsAllowed = !(isNoteEnterMode || playbackController()->isPlaying() || isMiddleButton);
```
```
 // move canvas
    if (!isNoteEnterMode || isMiddleButton) {
        m_view->moveCanvas(dragDelta.x(), dragDelta.y());
        m_isCanvasDragged = true;
    }
```
can you please check this.

Reply:
yes, good

Reply:
Should I update the pr?

Reply:
@Eism can you please merge it.